### PR TITLE
some bug fixes

### PR DESCRIPTION
**Pull request by alexandrupaler**

### CODE REVIEW

**Review by tylerrleblond — APPROVED** [review]

File: src/gates/parse_gates.cpp [hunk 1]
```diff
@@ -112,17 +113,31 @@ gates::CNOTAncillaPlacement determine_cnot_ancilla_placement(const std::vector<s
...
-        throw GateParseException{lstk::cat("Could not parse angle ", s, " as n*pi/m")};
+    if(split.size() != 2) {
+        //Is this an angle of the form 3*pi?
+        // not sure why but maybe we need something like
+        std::string ns = std::string(s) + "/1";
+
+        split = lstk::split_on(ns,"*pi/");
+        if(split.size() != 2)
+            throw GateParseException{lstk::cat("Could not parse angle ", s, " as n*pi/m")};
+    }
```
Comment (gwwatkin) [inline src/gates/parse_gates.cpp:136]: Yeah, this parser could for sure be more robust, starting from having a datatype to actually store fractions. Ideally we find a good header only library to parse and represent fractions

File: src/gates/parse_gates.cpp [hunk 2]
```diff
@@ -231,13 +266,14 @@ ParseGateResult parse_gate(std::string_view str_line)
...
 void GateStreamFromFile::advance_gate()
 {
-
     ParseGateResult maybe_gate = IgnoredInstruction{};
     while(!std::holds_alternative<gates::Gate>(maybe_gate))
     {
         std::string line;
         while(!gate_file_.eof() && line.size() ==0)
         {
+            std::cout << line;
```
Comment (gwwatkin) [inline src/gates/parse_gates.cpp:275]: I expect this to break the output when printing to stdout. I think it should be removed before merging

File: tests/gates/parse_gates.cpp [hunk 3]
```diff
@@ -12,11 +12,13 @@ TEST(parse_gates, rz_pi_over_1)
...
-    auto rz = std::get<gates::RZ>(g);
+//    ASSERT_TRUE(std::holds_alternative<gates::RZ>(g));
+    ASSERT_TRUE(std::holds_alternative<gates::BasicSingleQubitGate>(g));
+    auto rz = std::get<gates::BasicSingleQubitGate>(g);
     ASSERT_EQ(22, rz.target_qubit);
-    ASSERT_EQ(1, rz.pi_fraction.num);
-    ASSERT_EQ(1, rz.pi_fraction.den);
+    ASSERT_EQ(gates::BasicSingleQubitGate::Type::Z, rz.gate_type);
+//    ASSERT_EQ(1, rz.pi_fraction.num);
+//    ASSERT_EQ(1, rz.pi_fraction.den);
```
Comment (gwwatkin) [inline tests/gates/parse_gates.cpp:21]: comments can be removed

File: src/gates/parse_gates.cpp (hunk 1 above)
Comment (alexandrupaler) [reply src/gates/parse_gates.cpp:136]: there is a datatype for fractions, but was not sure how to use it for our purposes. I would not invest time in qasm parser, as i would expect somebody will eventually provide one open source on github

**Review by gwwatkin — APPROVED** [review]

**Pull request merged**